object/token.go

```diff
@@ -608,3 +611,59 @@ func GetTokenByUser(application *Application, user *User, scope string, host str
 	AddToken(token)
 	return token, nil
 }
+
+// Wechat Mini Program flow
+func GetWechatMiniProgramToken(application *Application, code string, providerName string, host string) (*Token, error) {
+	provider := GetProvider(util.GetId(providerName))
+	mpIdp := idp.NewWeChatMPIdProvider(provider.ClientId, provider.ClientSecret)
+	session, err := mpIdp.GetSeesionByCode(code)
+	if err != nil {
+		return nil, err
+	}
+	openId, unionId := session.Openid, session.Unionid
+	if openId == "" && unionId == "" {
+		return nil, errors.New("err: openid and unionid is empty")
+	}
+	user := getUserByOpenId(openId, unionId)
+	if user == nil {
+		//Add new user
+		user = &User{
+			Owner:             application.Organization,
+			Id:                util.GenerateId(),
+			Name:              fmt.Sprintf("wechat-%s", openId),
+			SignupApplication: application.Name,
+			WeChat:            openId,
+			Unionid:           unionId,
+			Type:              "normal-user",
+			CreatedTime:       util.GetCurrentTime(),
+			IsAdmin:           false,
+			IsGlobalAdmin:     false,
+			IsForbidden:       false,
+			IsDeleted:         false,
+		}
+		AddUser(user)
+	}
+
+	accessToken, refreshToken, err := generateJwtToken(application, user, "", "", host)
```

Comment:
I'm not familiar with the code and hope to confirm that is the JWT token has openId and unionId fields same as the wechat file implemented now.

```go
// idp/wechat.go
type WechatAccessToken struct {
	AccessToken  string `json:"access_token"`  //Interface call credentials
	ExpiresIn    int64  `json:"expires_in"`    //access_token interface call credential timeout time, unit (seconds)
	RefreshToken string `json:"refresh_token"` //User refresh access_token
	Openid       string `json:"openid"`        //Unique ID of authorized user
	Scope        string `json:"scope"`         //The scope of user authorization, separated by commas. (,)
	Unionid      string `json:"unionid"`       //This field will appear if and only if the website application has been authorized by the user's UserInfo.
}
```

Reply:
@leopku in idp/wechat.go, if the return has a unionID, then use the unionID, if only the openID, then use the openID, so I have the following code in this pr:
```go
existed, err := adapter.Engine.Where("wechat = ? OR wechat = ? OR unionid = ?", openId, unionId, unionId).Get(user)
```

Reply:
thanks for your explanation